app/src/main/java/com/android/sample/ui/utils/RecipeList.kt

```diff
@@ -234,29 +235,51 @@ private fun RecipeTitle(recipe: Recipe, modifier: Modifier) {
  * @param recipe the recipe to like.
  */
 @Composable
-fun TopCornerLikeButton(recipe: Recipe, userViewModel: UserViewModel) {
+fun TopCornerLikeButton(recipe: Recipe, userViewModel: UserViewModel, removeLikedRecipe: Boolean) {
   var recipeUnlike: Boolean by remember { mutableStateOf(false) }
-
+  var recipeDelete: Boolean by remember { mutableStateOf(false) }
   Icon(
-      imageVector = if (!recipeUnlike) Icons.Filled.Favorite else Icons.Filled.FavoriteBorder,
+      imageVector =
+          if (removeLikedRecipe)
+              (if (!recipeUnlike) Icons.Filled.Favorite else Icons.Filled.FavoriteBorder)
+          else Icons.Filled.Delete,
       contentDescription = RECIPE_FAVORITE_ICON_CONTENT_DESCRIPTION,
       modifier =
           Modifier.padding(4.dp).testTag(RECIPE_FAVORITE_ICON_TEST_TAG).clickable {
-            recipeUnlike = true
+            if (removeLikedRecipe) {
+              recipeUnlike = true
+            } else {
+              recipeDelete = true
+            }
           },
```

Comment:
It might be a good idea to increase the icon size for better visibility, perhaps to 22 or 24 dp. Here's a potential way to implement it:

```suggestion
      Modifier.size(24.dp)
```
Let me know what you think!